benchmark/fs/promises-access.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const common = require('../common');
+const { access, async } = require('fs');
+const { promisify } = require('util');
+const Countdown = require('../../test/common/countdown');
+
+const bench = common.createBenchmark(main, {
+  n: [20e4],
+  method: ['legacy', 'promisify', 'promise']
+});
+
+function useLegacy(n) {
+  const countdown = new Countdown(n, () => {
+    bench.end(n);
+  });
+  bench.start();
+  for (let i = 0; i < n; i++) {
+    access(__filename, () => countdown.dec());
```

Comment:
I am not convinced that running 2e5 calls in the same tick is actually a good way of benchmarking this. I think it would be better to either use a recursive strategy to see the individual call length or a batched recursive call that batches up to lets say 500 calls together.
This applies to all these benchmarks.

Reply:
I agree, ideally if we can get a more applicative benchmark that'd be the best

Reply:
In particular, the promises version always queue the callback to run later every time here which would be disproportionately represented in the above benchmark (I think)